### PR TITLE
Feature/bar chart

### PR DESCRIPTION
**Pull request by richardiskra**
Added functionality to the bar chart. Users can select a curriculum to see it's progress. Colors changed to match the status.

### CODE REVIEW

File: src/app/routes/boom/boom.component.ts [hunk 1]
```diff
@@ -31,46 +53,58 @@ export class BoomComponent implements OnInit {
...
-  // ponut chart
+  // donut chart
   public doughnutChartLabels: string[] = ['Completed', 'Missed'];
   public doughnutChartData: number[] = this.getProgres();
   public doughnutChartType = 'doughnut';
-
+  public donutColors = [{backgroundColor: ['rgb(0, 189, 0)', 'rgb(230, 230, 90)']}];
 
   /**
    * determines how many tasks have been completed and how many have been missed
```
Comment (YamasakiRhys) [inline src/app/routes/boom/boom.component.ts:82]: Don't forget the @author annotation

File: src/app/routes/boom/boom.component.ts [hunk 2]
```diff
@@ -56,13 +56,24 @@ export class BoomComponent implements OnInit {
     { data: [], label: 'Completed' },
     { data: [], label: 'Missed' }
   ];
-  public barColors = [{backgroundColor: ['rgb(0, 189, 0)', 'rgb(230, 230, 90)']}];
+
+  public barColors: Array<any> = [
+    { // correct color
```
Comment (YamasakiRhys) [inline src/app/routes/boom/boom.component.ts]: Move comment to newline

File: src/app/routes/boom/boom.component.ts [hunk 3]
```diff
@@ -80,12 +80,12 @@ export class BoomComponent implements OnInit {
    */
   getProgres() {
     const prog: number[] = [0, 0];
-    for (let k = 0; k < this.data.length; k++) {
-      for (let i = 0; i < this.data[k].curriculum.length; i++) {
-        for (let j = 0; j < this.data[k].curriculum[i].tasks.length; j++) {
-          if (this.data[k].curriculum[i].tasks[j].status === 1) {
+    for (let k = 0; k < this.data.length; k++) { // for each curriculum
```
Comment (YamasakiRhys) [inline src/app/routes/boom/boom.component.ts]: Have these comments on a newline.  Looks kind of awkward

**Review by YamasakiRhys — DISMISSED** [review]

File: src/app/routes/boom/boom.component.ts [hunk 4]
```diff
@@ -31,56 +53,107 @@ export class BoomComponent implements OnInit {
...
+      for (let i = 0; i < this.data[k].curriculum.length; i++) {
+        // for each task
+        for (let j = 0; j < this.data[k].curriculum[i].tasks.length; j++) {
+          // if the task was completed
+          if (this.data[k].curriculum[i].tasks[j].status === 1) {
+            ++prog[0];
+          // if the task was missed
+          } else if (this.data[k].curriculum[i].tasks[j].status === 2) {
+            ++prog[1];
+          }
```
Comment (danielsidhu) [inline src/app/routes/boom/boom.component.ts:98]: good code. good styling

File: src/app/routes/boom/boom.component.ts [hunk 5]
```diff
@@ -31,56 +53,107 @@ export class BoomComponent implements OnInit {
...
+  }
+
+  ngOnInit() {
+    this.getWeeklyProgress(0);
   }
 }
+
+/**
+ * @author Richard Iskra | Obosa Nosa Igeibaro | Eddie Grays
+ */
```
Comment (danielsidhu) [inline src/app/routes/boom/boom.component.ts:159]: good.

**Review by danielsidhu — APPROVED** [review]
If you are ready to merge and conflicts that Rhys had are resolved I will merge soon

**Review by ChineduOzodi — APPROVED** [review]
Looks good.

**Comment by ChineduOzodi** [comment]
@richardiskra Are you done with this branch so we can merge and delete it?

**Comment by richardiskra** [comment]
Yes, this branch is done

**Pull request merged**